### ISSUE TITLE
ci: Run Node.js and Python SDK tests on Windows and macOS as part of merges

### ISSUE THREAD

**Issue by justinvp** [pulumi/pulumi#15295]
Currently, we only run these on Linux. There is Windows-specific code that we should be testing.

**Comment by julienp** [comment]
My understanding is that we never run any macos/windows tests on PRs, and only do so on merge.

Even for community PRs where we use `/run-acceptance-tests`, we only run the "integration" tests, but not the "acceptance tests" (which are the macos/windows) tests.

Using the label `ci/test` runs the acceptance tests on a PR.

**Comment by Frassle** [comment]
I think the point is some of the tests are only running on linux for both PR and Merge jobs.

**Comment by julienp** [comment]
> I think the point is some of the tests are only running on linux for both PR and Merge jobs.

Ah, understood. So is the current expected workflow to use the `ci/test` label when we suspect it's a good idea to run windows/macos tests, but otherwise defer them to merge?

**Comment by Frassle** [comment]
Yup, generally it's more cost than value running on macOS and Windows for every push to a PR but we always want to check them before merge.
